cmd/influxd/run/server.go

```diff
@@ -299,7 +302,23 @@ func (s *Server) appendHTTPDService(c httpd.Config) {
 	ss := storage.NewStore(s.TSDBStore, s.MetaClient)
 	srv.Handler.Store = ss
 	if s.config.HTTPD.FluxEnabled {
-		srv.Handler.Controller = control.NewController(s.MetaClient, reads.NewReader(ss), authorizer, c.AuthEnabled, s.PointsWriter, s.Logger)
+		storageDep, err := influxdb2.NewDependencies(s.MetaClient, reads.NewReader(ss), authorizer, c.AuthEnabled, s.PointsWriter)
+		if err != nil {
+			panic(err)
```

Comment:
Would it be very disruptive to have this method return `error` instead of panicking?

Reply:
Not very. Fixed.